react-components/src/architecture/concrete/observations/ObservationsView.ts

```diff
@@ -0,0 +1,161 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { Box3, type Plane, Vector3 } from 'three';
+import { type ObservationsDomainObject } from './ObservationsDomainObject';
+import { GroupThreeView } from '../../base/views/GroupThreeView';
+import {
+  CDF_TO_VIEWER_TRANSFORMATION,
+  Overlay3DCollection,
+  type OverlayInfo,
+  type CustomObjectIntersectInput,
+  type CustomObjectIntersection
+} from '@cognite/reveal';
+import { type DomainObjectIntersection } from '../../base/domainObjectsHelpers/DomainObjectIntersection';
+import { Changes } from '../../base/domainObjectsHelpers/Changes';
+import { type DomainObjectChange } from '../../base/domainObjectsHelpers/DomainObjectChange';
+import { createObservationIntersection, type Observation } from './types';
+import { ClosestGeometryFinder } from '../../base/utilities/geometry/ClosestGeometryFinder';
+import { getColorFromStatus } from './color';
+
+type ObservationCollection = Overlay3DCollection<Observation>;
+
+export class ObservationsView extends GroupThreeView<ObservationsDomainObject> {
+  private readonly _overlayCollection: ObservationCollection = new Overlay3DCollection([]);
+
+  protected override calculateBoundingBox(): Box3 {
+    return this._overlayCollection
+      .getOverlays()
+      .reduce(
+        (box, overlay) => (overlay.getVisible() ? box.expandByPoint(overlay.getPosition()) : box),
+        new Box3()
+      );
+  }
+
```

Comment:
I will prefer this code instead. More readable in my eyes  

```
protected override calculateBoundingBox(): Box3 {
    const boundingBox = new Box3().makeEmpty();
    for (const overlay of this._overlayCollection.getOverlays()) {
      if (overlay.getVisible()) {
        boundingBox.expandByPoint(overlay.getPosition());
      }
    }
    return boundingBox;

```

Reply:
Okay, fine :+1: